### PR TITLE
feat: update redux store and UI for system + user logs

### LINKED ISSUE #14432: Merge system logs and user generated console logs to the Logs tab

### LINKED ISSUE #14433: Update UI of user generated console logs to appsmith design system

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description

Updated the redux store to handle the UI changes for the new debugger experience. Basically added two new keys, one for mentioning whether the generated source is the user's console statement or the system logs.

Link for final designs: 
1. [Figma](https://www.figma.com/file/X03G0OsEtvH9JHDhU8XREy/JS-Editor-Improvements?node-id=2186%3A112955)
2. [Zeplin](https://zpl.io/g8X1pq0)

Fixes #14432 
Fixes #14433 

## Type of change

- New feature (non-breaking change which adds functionality)

**Test plan**
Test plan coverage is present in PR (https://github.com/appsmithorg/appsmith/issues/15893)

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by arunvjn** [comment]
@ayushpahwa We are using a dropdown to show all/only console logs/only system logs in this PR and it is different from the figma designs which uses checkboxes. Is it intentional ? Also, the dropdown has different height than the filter input. Can we make it the same ?

**Comment by arunvjn** [comment]
Screenshot 2022-08-25 at 09 54 07

Also, the dropdown bleeds out of the debugger in canvas. I am on a 13inch mac book standard resolution.

**Review by albinAppsmith — APPROVED** [review]

File: app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx [hunk 1]
```diff
@@ -31,14 +32,27 @@ type Props = {
...
+const LOGS_FILTER_OPTIONS = (theme: DefaultTheme) => [
   {
     label: "All",
     value: "",
   },
-  { label: "Success", value: Severity.INFO },
-  { label: "Warnings", value: Severity.WARNING },
-  { label: "Errors", value: Severity.ERROR },
+  {
+    label: "Errors",
```
Comment (arunvjn) [inline app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx:41]: Can we move these labels to messages.ts ?

File: app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx [hunk 2]
```diff
@@ -47,8 +61,9 @@ function DebbuggerLogs(props: Props) {
   const filteredLogs = useFilteredLogs(searchQuery, filter);
   const { next, paginatedData } = usePagination(filteredLogs);
   const listRef = useRef<HTMLDivElement>(null);
+  const theme = useTheme();
   const selectedFilter = useMemo(
-    () => LOGS_FILTER_OPTIONS.find((option) => option.value === filter),
+    () => LOGS_FILTER_OPTIONS(theme).find((option) => option.value === filter),
     [filter],
```
Comment (arunvjn) [inline app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx:67]: Shouldn't theme be a dependency here ?

File: app/client/src/components/editorComponents/Debugger/FilterHeader.tsx [hunk 3]
```diff
@@ -21,34 +21,31 @@ const Wrapper = styled.div`
...
   }
 
   .debugger-filter {
-    border: none;
-    box-shadow: none;
-    width: 110px;
-    height: 28px;
-    margin-top: ${(props) => props.theme.spaces[1]}px;
+    /* border: none;
+    box-shadow: none; */
```
Comment (arunvjn) [inline app/client/src/components/editorComponents/Debugger/FilterHeader.tsx:34]: Can we remove these unused css ?

File: app/client/src/components/editorComponents/Debugger/LogItem.tsx [hunk 4]
```diff
@@ -19,9 +24,16 @@ import {
...
+  display: flex;
+  align-items: center;
+`;
+
 const Wrapper = styled.div<{ collapsed: boolean }>`
-  padding: 9px 30px;
+
   display: flex;
+  flex-direction: column;
+  padding: 9px 8px 9px 16px;
```
Comment (arunvjn) [inline app/client/src/components/editorComponents/Debugger/LogItem.tsx:36]: Shouldn't spacing be in multiples of 4 ?

File: app/client/src/components/editorComponents/Debugger/hooks/debuggerHooks.ts [hunk 5]
```diff
@@ -24,15 +24,26 @@ export const useFilteredLogs = (query: string, filter?: any) => {
...
   }
 
   if (query) {
     logs = logs.filter((log) => {
-      if (log.source?.name)
-        return (
-          log.source?.name.toUpperCase().indexOf(query.toUpperCase()) !== -1
-        );
+      if (log.source?.name) {
+        if (log.source?.name.toUpperCase().indexOf(query.toUpperCase()) !== -1)
```
Comment (arunvjn) [inline app/client/src/components/editorComponents/Debugger/hooks/debuggerHooks.ts:35]: Nit: Optional chaining is not required here

File: app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx (hunk 2 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx]: Updated

File: app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx (hunk 1 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/Debugger/DebuggerLogs.tsx]: Updated

File: app/client/src/components/editorComponents/Debugger/FilterHeader.tsx (hunk 3 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/Debugger/FilterHeader.tsx]: Updated

File: app/client/src/components/editorComponents/Debugger/LogItem.tsx (hunk 4 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/Debugger/LogItem.tsx]: Updated

File: app/client/src/components/editorComponents/Debugger/hooks/debuggerHooks.ts (hunk 5 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/Debugger/hooks/debuggerHooks.ts]: Updated

**Pull request merged**